### PR TITLE
AF-2848: Replacement of org.jboss.spec.javax.* deps

### PR DESCRIPTION
**Pull request by mbiarnes**
(cherry picked from commit 932b99f586ecc4b8e6aeeab130987831cd829d48)

**Thank you for submitting this pull request**

[droolsjbpm-build-bootstrap](https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1626)
[kie-soup](https://github.com/kiegroup/kie-soup/pull/205)
[droolsjbpm-knowledge](https://github.com/kiegroup/droolsjbpm-knowledge/pull/521)
[drools](https://github.com/kiegroup/drools/pull/3628)
[optaplanner](https://github.com/kiegroup/optaplanner/pull/1333)
[appformer](https://github.com/kiegroup/appformer/pull/1143)
[jbpm](https://github.com/kiegroup/jbpm/pull/1917)
[kie-jpmml-integration](https://github.com/kiegroup/kie-jpmml-integration/pull/54)
[jbpm-wb](https://github.com/kiegroup/jbpm-wb/pull/1489)

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by mbiarnes** [comment]
Jenkins retest this

**Comment by mbiarnes** [comment]
Jenkins retest this

**Comment by mbiarnes** [comment]
Jenkins retest this

**Comment by mbiarnes** [comment]
Jenkins retest this

File: uberfire-api/src/main/java/org/uberfire/java/nio/base/version/VersionRecord.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    String author();
+
+    String email();
+
+    String comment();
+
+    Date date();
+
+    String uri();
+}
```
Comment (mareknovotny) [inline uberfire-api/src/main/java/org/uberfire/java/nio/base/version/VersionRecord.java:34]: Remove the class here or from uberfire-nio2-model module

File: pom.xml [hunk 2]
```diff
@@ -224,6 +224,7 @@
                            and there is very little chance this will get ever fixed. -->
                       <ignoreClass>org.w3c.dom.ElementTraversal</ignoreClass>
                       <ignoreClass>org.w3c.dom.UserDataHandler</ignoreClass>
+                      <ignoreClass>org.uberfire.java.nio.base.version.VersionRecord</ignoreClass>
```
Comment (mareknovotny) [inline pom.xml:227]: the ignoreClass should be explained and we usually do not do this for classes from this repository or from dependency we directly control. Please remove this

**Comment by mareknovotny** [comment]
please rebase this as there were a breakage in master due  https://github.com/kiegroup/appformer/pull/1147

**Comment by mbiarnes** [comment]
Jenkins retest this

**Comment by mbiarnes** [comment]
This PR will be closed since Tiago Bento and William Siqueira are working on the same issue:
https://github.com/kiegroup/appformer/pull/1163

**Pull request closed without merging**